pkg/gds/models/v1/models_test.go

```diff
@@ -448,6 +449,79 @@ func TestVeriedContacts(t *testing.T) {
 	require.Len(t, contacts, 2)
 }
 
+func TestNewCertificateRequest(t *testing.T) {
+	// Should not be able to create a certificate request with a nil vasp
+	_, err := NewCertificateRequest(nil)
+	require.Error(t, err)
+
+	// If the name does not exist, the default value should be populated
+	vasp := &pb.VASP{
+		Id:         "b5841869-105f-411c-8722-4045aad72717",
+		CommonName: "charlieVASP",
+		Entity:     &ivms101.LegalPerson{},
+	}
+	cr, err := NewCertificateRequest(vasp)
+	require.NoError(t, err)
+	require.Equal(t, vasp.Id, cr.Vasp)
+	require.Equal(t, vasp.CommonName, cr.CommonName)
+	require.Contains(t, cr.Params, "organizationName")
+	require.Equal(t, "TRISA Production", cr.Params["organizationName"])
+
+	vasp.Entity.Name = &ivms101.LegalPersonName{
+		NameIdentifiers: []*ivms101.LegalPersonNameId{
+			{
+				LegalPersonName:               "Charlie Inc.",
+				LegalPersonNameIdentifierType: ivms101.LegalPersonShort,
+			},
+		},
+	}
+	cr, err = NewCertificateRequest(vasp)
+	require.NoError(t, err)
+	require.Contains(t, cr.Params, "organizationName")
+	require.Equal(t, "TRISA Production", cr.Params["organizationName"])
+
+	// Valid organization name, unspecified location info
+	vasp.Entity.Name.NameIdentifiers[0].LegalPersonNameIdentifierType = ivms101.LegalPersonLegal
+	cr, err = NewCertificateRequest(vasp)
+	require.NoError(t, err)
+	require.Contains(t, cr.Params, "organizationName")
+	require.Equal(t, "Charlie Inc.", cr.Params["organizationName"])
+	require.Contains(t, cr.Params, "localityName")
+	require.Equal(t, "Menlo Park", cr.Params["localityName"])
+	require.Contains(t, cr.Params, "stateOrProvinceName")
+	require.Equal(t, "California", cr.Params["stateOrProvinceName"])
+	require.Contains(t, cr.Params, "countryName")
+	require.Equal(t, "US", cr.Params["countryName"])
+
+	vasp.Entity.GeographicAddresses = []*ivms101.Address{
+		{
+			TownLocationName:   "Toronto",
+			CountrySubDivision: "Ontario",
+		},
```

Comment:
Could we change the partial info test to having Country but not town or state? This is the common case in live data, so I'd like to make sure we're protecting from regression with that case.